src/statement.js

```diff
@@ -424,17 +442,23 @@ pp.parseVarId = function(decl) {
 // Parse a function declaration or literal (depending on the
 // `isStatement` parameter).
 
-pp.parseFunction = function(node, isStatement, allowExpressionBody) {
+pp.parseFunction = function(node, isStatement, allowExpressionBody, isAsync) {
   this.initFunction(node)
-  if (this.options.ecmaVersion >= 6)
+  if (this.options.ecmaVersion >= 6 && !isAsync)
     node.generator = this.eat(tt.star)
-  var oldInGen = this.inGenerator
+  if (this.options.ecmaVersion >= 8)
+    node.async = !!isAsync
+  if (isStatement && this.type === tt.name)
```

Comment:
```js
// Also, the function declaration at `export default` can be omitted their name.
export default function() {}
export default async function() {}
```

Above functions are function declarations (those are hoisted) in the language spec, but Acorn addresses those as function expressions. :confused: 
Also, we seem to not need to consider `await` and `yield` as function names for modules because `await` and `yield` are keywords in modules.
So, I can simply drop `this.type === tt.name`.